argilla/src/argilla/settings/_io/_hub.py

```diff
@@ -0,0 +1,167 @@
+        if dtype == "string":
+            return FeatureType.TEXT
+        elif dtype in ["int32", "int64"]:
+            return FeatureType.INT
+        elif dtype in ["float32", "float64"]:
+            return FeatureType.FLOAT
+    elif hf_type == "Image":
+        return FeatureType.IMAGE
+    elif hf_type == "ClassLabel":
+        return FeatureType.LABEL
+    else:
+        warnings.warn(f"Unsupported feature type. hf_type: {hf_type}, dtype: {dtype}")
+
+
+def _is_chat_feature(sub_features):
+    """Check if the sub_features correspond to an rg.ChatField."""
+    return (
+        "content" in sub_features
+        and "role" in sub_features
+        and sub_features["content"]["_type"] == "Value"
+        and sub_features["role"]["_type"] == "Value"
+        and sub_features["content"].get("dtype") == "string"
+        and sub_features["role"].get("dtype") == "string"
+    )
+
+
+def _define_settings_from_features(features: Union[List[Dict], Dict[str, Any]]) -> "Settings":
+    """Define the argilla settings from the features of a dataset.
+
+    Parameters:
+        features (Dict[str, Dict[str, Any]): The features of the dataset.
+
+    Returns:
+        rg.Settings: The settings defined from the features.
+    """
+
+    from argilla import ImageField, LabelQuestion, TextQuestion, Settings, TextField
+
+    fields = []
+    questions = []
+    metadata = []
+    mapping = {}
+
+    for name, feature in features.items():
+        feature_type = _map_feature_type(feature)
+
+        if feature_type == FeatureType.CHAT:
+            # TODO: Implement chat support to create `rg.ChatField`
+            # fields.append(rg.ChatField(name=f"{name}_field"))
+            # mapping[name] = f"{name}_field"
+            pass
+
+        elif feature_type == FeatureType.TEXT:
+            fields.append(TextField(name=f"{name}_field"))
+            questions.append(TextQuestion(name=f"{name}_question"))
+            mapping[name] = (f"{name}_field", f"{name}_question")
+
+        elif feature_type == FeatureType.IMAGE:
+            fields.append(ImageField(name=f"{name}_field"))
+            mapping[name] = f"{name}_field"
+
+        elif feature_type == FeatureType.LABEL:
+            fields.append(TextField(name=f"{name}_field"))
+            questions.append(
+                LabelQuestion(
+                    name=f"{name}_question",
+                    labels={str(i): label for i, label in enumerate(feature.get("names", []))},
+                )
+            )
+            mapping[name] = (f"{name}_field", f"{name}_question")
+
+        elif feature_type in [FeatureType.INT, FeatureType.FLOAT]:
+            # TODO: Implement metadata support to create `rg.FloatMetadata` or `rg.IntMetadata`
+            pass
+        else:
+            warnings.warn(f"Feature '{name}' has an unsupported type. Skipping. Feature type: {feature_type}")
+
+    settings = Settings(fields=fields, questions=questions, metadata=metadata, mapping=mapping)
```

Comment:
Maybe we can add this if no questions were mapped (with a warning). So we can remove the TextQuestion for text features. What do you think?

```suggestion
    if not questions:
        questions.append(rg.TextQuestion(name="comment",...)
    settings = Settings(fields=fields, questions=questions, metadata=metadata, mapping=mapping)
```

Reply:
Makes sense